src/wallet/walletdb.h

```diff
@@ -58,13 +61,16 @@ class CHDChain
     {
         READWRITE(this->nVersion);
         READWRITE(nExternalChainCounter);
+        if (this->nVersion >= VERSION_HD_CHAIN_SPLIT)
```

Comment:
Where do you see the need to set the `nInternalChainCounter` for old versions here? Old Versions will always have the `nInternalChainCounter` set the `0` over `SetNull()` during the constructor call.

A new chain (which then will enable `VERSION_HD_CHAIN_SPLIT`) can only be created when a new HD master key will be set which again will enable `FEATURE_HD_SPLIT`.